src/components/ItemsList.js

```diff
@@ -23,13 +32,33 @@ const ItemsList = () => {
     { idField: 'documentId' },
   );
 
-
   const markItemAsPurchased = (index) => {
     const { items, documentId } = shoppingList[0];
-
-    items[index].lastPurchasedOn = items[index].lastPurchasedOn
-      ? null
-      : Date.now();
+    const { lastPurchasedOn } = items[index];
+    const shoppingItemObject = items[index];
+    if (lastPurchasedOn === null) {
```

Comment:
I think the lastPurchasedOn date can never be null. It can be set once the user adds an item to the list and then on the subsequent purchase that value can be used. I could be wrong but thats how I see it. 😄

Reply:
ok thanks for the review, the lastPurchasedOn is indeed null until a user makes first purchase(when they check an item). They can add an item as a todo list without outrightly buying the item, lastPurchasedOn is null at that point, then when they tick the item as bought, lastPurchasedOn will be recorded as the time when they ticked the box.

Reply:
Yes, I agree @Amaka202. Initially, the `lastPurchasedOn` should be null, as it also was part of the previous issue's AC.

Reply:
That's right. Thanks for the clarification 👍🏻